src/routers/products.router.ts

```diff
@@ -12,10 +12,10 @@ import { paginateMiddleware } from '../middlewares/paginate.middleware';
 
 const productsRouter: Router = Router();
 
+productsRouter.use(['/popular', '/limited-edition', '/'], paginateMiddleware);
```

Comment:
Check the comment for the other PR, this need to be changed to the best-practice I described there.